openmetadata-ui/src/main/resources/ui/src/components/ActivityFeed/ActivityFeedCardV2/ActivityFeedCardV2.tsx

```diff
@@ -63,31 +77,124 @@ const ActivityFeedCardV2 = ({
+      updateAnnouncementThreads && updateAnnouncementThreads();
+    } else {
+      updateFeed(feed.id, post.id, !isPost, patch);
+    }
+
     setIsEditPost(!isEditPost);
   };
 
+  const updateThreadHandler = async (
+    threadId: string,
+    postId: string,
+    isThread: boolean,
+    data: Operation[]
+  ): Promise<void> => {
+    const callback = () => {
+      return;
+    };
+    await updateThreadData(threadId, postId, isThread, data, callback);
+    fetchUpdatedThread(threadId);
+  };
+  const handleAnnouncementUpdate = async (
+    title: string,
+    announcement: AnnouncementDetails
+  ) => {
+    const existingAnnouncement = {
+      ...feed,
+      announcement: feed.announcement,
+    };
+
+    const updatedAnnouncement = {
+      ...feed,
+      message: title,
+      announcement,
+    };
+
+    const isAnnouncementTimeUpdated =
+      _.isEqual(
+        existingAnnouncement?.announcement?.startTime,
+        updatedAnnouncement?.announcement?.startTime * 1000
+      ) &&
+      _.isEqual(
+        existingAnnouncement?.announcement?.endTime,
+        updatedAnnouncement?.announcement?.endTime * 1000
+      );
+
+    const patch = compare(existingAnnouncement, updatedAnnouncement);
+
+    if (!isEmpty(patch)) {
+      updateThreadHandler(feed.id, feed.id, true, patch);
+      if (isAnnouncementTab) {
+        updateAnnouncementThreads && updateAnnouncementThreads(); // if its Announcement tab in service page
+      } else {
+        !isAnnouncementTimeUpdated && // refetch new announcements only if announcements timings are updated
+          updateAnnouncementThreads &&
+          updateAnnouncementThreads();
+      }
+    }
+
+    setIsEditAnnouncement(false);
+  };
+  const repliesPostAvatarGroup = useMemo(() => {
+    return (
+      <Avatar.Group>
+        {(feed.posts ?? []).map((u) => (
+          <ProfilePicture
+            avatarType="outlined"
+            key={u.id}
+            name={u.from}
+            width="18"
+          />
+        ))}
+      </Avatar.Group>
+    );
+  }, [feed.posts]);
+
   return (
     <div
       className={classNames(
         'feed-card-v2-container p-sm',
         {
-          active: isActive,
+          active: isActive && feed.type !== ThreadType.Announcement,
+          'announcement-active':
+            isActive && feed.type === ThreadType.Announcement,
+          'announcement-gap': feed.type === ThreadType.Announcement,
         },
         className
       )}>
-      <div
-        className={classNames('feed-card-v2-sidebar', {
-          'feed-card-v2-post-sidebar': isPost,
-        })}>
-        <ProfilePicture
-          avatarType="outlined"
-          name={post.from}
-          size={isPost ? 28 : 30}
-          width={isPost ? '28' : '30'}
-        />
-      </div>
-      <Row className="w-full" gutter={[0, 10]}>
+      {feed.type === ThreadType.Announcement &&
+        !isPost &&
+        componentsVisibility.showRepliesContainer && (
```

Comment:
Let's pull out complex conditions and try to simplify those in a better way

Reply:
done, extracted these conditions out and simplified it.